test/correctness/mul_div_mod.cpp

```diff
@@ -569,11 +573,24 @@ int main(int argc, char **argv) {
         }
     }
 
-    Halide::Internal::ThreadPool<bool> pool;
+    size_t num_threads = Halide::Internal::ThreadPool<bool>::num_processors_online();
+    if (target.has_feature(Target::OpenCL)) {
```

Comment:
ooh, could the bug be a race condition somewhere inside the compiler between independent compilations?

Reply:
yeah, that's one theory. I don't see any obvious issues that make this likely, though. If this seems to clear it up, some targeting investigation of this theory might be worthwhile.

Reply:
Maybe we should just put a mutex around kernel compilation as a speculative fix? I doubt this would be a performance issue in any real case.

Reply:
You mean in case the race condition is inside our opencl runtime at kernel compilation term? I was worried it was inside libHalide, and we don't want to put a mutex around all of compilation.

Reply:
I guess it could be, but these tests are only failing on OpenCL and CUDA. They're still parallelized for x86, ARM, Hexagon and we haven't seen any flakes on those targets. So I suspect the issue is in the runtime around initialization of the kernels.